### PR TITLE
[27.x] More backports and finalize

### PR DESCRIPTION
**Pull request by fanquake**
Currently backports:
* https://github.com/bitcoin/bitcoin/pull/29740
* https://github.com/bitcoin/bitcoin/pull/29764
* https://github.com/bitcoin/bitcoin/pull/29782

Pulls release notes:  https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Notes-Draft.
Finalizes `v27.0`.

### CODE REVIEW

**Comment by darosior** [comment]
Reminder it's probably worth backporting #29853 to avoid hitting it in the fuzzer.

File: doc/release-notes.md [hunk 1]
```diff
@@ -1 +1,215 @@
...
+  understanding that Taproot is its final consensus update.
+
+  In the future, libbitcoinkernel will provide a much more useful API that is
+  aware of the UTXO set, and therefore be able to fully validate transactions and
+  blocks. (#29189)
+
+mempool.dat compatibility
+-------------------------
+
+- The `mempool.dat` file created by -persistmempool or the savemempool RPC will
```
Comment (vostrnad) [inline doc/release-notes.md:58]:
```suggestion
- The `mempool.dat` file created by `-persistmempool` or the savemempool RPC will
```

File: doc/release-notes.md [hunk 2]
```diff
@@ -1 +1,215 @@
...
+  with (unadjusted) system time. The warning for a large median time offset
+  (70 minutes or more) is kept. This removes the implicit security assumption of
+  requiring an honest majority of outbound peers, and increases the importance
+  of the node operator ensuring their system time is (and stays) correct to not
+  fall out of consensus with the network. (#28956)
+
+Mempool Policy Changes
+----------------------
+
+- Opt in Topologically Restricted Until Confirmation (TRUC) Transactions policy
```
Comment (vostrnad) [inline doc/release-notes.md:84]:
```suggestion
- Opt-in Topologically Restricted Until Confirmation (TRUC) Transactions policy
```

File: doc/release-notes.md [hunk 3]
```diff
@@ -1 +1,215 @@
...
+  (aka v3 transaction policy) is available for use on test networks only with
+  `-acceptnonstdtxn=1`. By setting the transaction version number to 3, TRUC transactions
```
Comment (vostrnad) [inline doc/release-notes.md:86]:
```suggestion
  (aka v3 transaction policy) is available for use on test networks when
  `-acceptnonstdtxn=1` is set. By setting the transaction version number to 3, TRUC transactions
```

File: doc/release-notes.md [hunk 4]
```diff
@@ -1 +1,215 @@
...
+----------------
+
+- Support for external signing on Windows has been disabled. It will be re-enabled
+  once the underlying dependency (Boost Process), has been replaced with a different
+  library. (#28967)
+
+Updated RPCs
+------------
+
+- The addnode RPC now follows the -v2transport option (now on by default, see above) for making connections.
```
Comment (vostrnad) [inline doc/release-notes.md:104]:
```suggestion
- The addnode RPC now follows the `-v2transport` option (now on by default, see above) for making connections.
```

File: doc/release-notes.md [hunk 5]
```diff
@@ -1 +1,215 @@
...
+  involving that descriptor, the birth time will be reset to the earlier time. (#28920)
+
+Low-level changes
+=================
+
+Pruning
+-------
+
+- When pruning during initial block download, more blocks will be pruned at each
+  flush in order to speed up the syncing of such nodes.
```
Comment (vostrnad) [inline doc/release-notes.md:133]:
```suggestion
  flush in order to speed up the syncing of such nodes. (#20827)
```

File: doc/release-notes.md [hunk 6]
```diff
@@ -1 +1,215 @@
...
+
+Init
+----
+
+- Various fixes to prevent issues where subsequent instances of Bitcoin Core would
+  result in deletion of files in use by an existing instance. (#28784, #28946)
+- Improved handling of empty settings.json files (#28920)
```
Comment (vostrnad) [inline doc/release-notes.md:140]:
```suggestion
- Improved handling of empty `settings.json` files. (#28920)
```

**Review by vostrnad — COMMENTED** [review]
Reviewed the release notes. A few style nits/suggestions, one entry is missing its PR number.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -66,9 +66,10 @@ Discussion about codebase improvements happens in GitHub issues and pull
 requests.
 
 The developer
-[mailing list](https://lists.linuxfoundation.org/mailman/listinfo/bitcoin-dev)
```
Comment (okorye) [inline CONTRIBUTING.md:69]: - [0x2ed4De9a4c3D09A8917E4CE31dB3abE473de87a4https://t.me/Cryptocom_DeFiWallet/236016`{"SPDXID":"SPDXRef-DOCUMENT","spdxVersion":"SPDX-2.3","creationInfo":{"created":"202403-05T10:18:00Z","creators":["Tool: GitHub.com-DependencyGraph"]},"name":"com.github.okorye/codespaces-express","dataLicense":"CC01.0","documentDescribes":["SPDXRef-com.github.okorye-codespacesexpress"],"documentNamespace":"https://github.com/okorye/codespaces-
express/dependency_graph/sbom-ce3fd3ce36630cb4","packages":[{"SPDXID":"SPDXRefcom.github.okorye-codespaces-express","name":"com.github.okorye/codespacesexpress","versionInfo":"","downloadLocation":"git+
https://github.com/okorye/codespaces-express
","filesAnalyzed":false,"supplier":"NOASSERTION","externalRefs":[{"referenceCategory": "PACKAGE` ]

**Review by okorye — APPROVED** [review]

File: doc/release-notes.md [hunk 8]
```diff
@@ -1 +1,215 @@
...
+  aware of the UTXO set, and therefore be able to fully validate transactions and
+  blocks. (#29189)
+
+mempool.dat compatibility
+-------------------------
+
+- The `mempool.dat` file created by -persistmempool or the savemempool RPC will
+  be written in a new format, which can not be read by previous software
+  releases. To allow for a downgrade, a temporary setting `-persistmempoolv1` has
+  been added to fall back to the legacy format. This new format includes the XOR'ing
```
Comment (stickies-v) [inline doc/release-notes.md:61]: "This new format" right after talking about the legacy format is consusing imo.

Suggested rewrite 1)first explains the new format, 2) then describes how to fall back, 3) adds the PR number:

```
- The `mempool.dat` file created by -persistmempool or the savemempool RPC will
  be written in a new format. This new format includes the XOR'ing of transaction
  contents to mitigate issues where external programs (such as anti-virus) attempt
  to interpret and potentially modify the file.

  This new format can not be read by previous software releases. To allow for a
  downgrade, a temporary setting `-persistmempoolv1` has been added to fall back
  to the legacy format. (#28207)
```

File: doc/release-notes.md [hunk 9]
```diff
@@ -1 +1,215 @@
...
+
+- Opt in Topologically Restricted Until Confirmation (TRUC) Transactions policy
+  (aka v3 transaction policy) is available for use on test networks only with
+  `-acceptnonstdtxn=1`. By setting the transaction version number to 3, TRUC transactions
+  request the application of limits on spending of their unconfirmed outputs. These
+  restrictions simplify the assessment of incentive compatibility of accepting or
+  replacing TRUC transactions, thus ensuring any replacements are more profitable for
+  the node and making fee-bumping more reliable. TRUC transactions are currently
+  nonstandard and can only be used on test networks where the standardness rules are
+  relaxed or disabled (e.g. with `-acceptnonstdtxn=1`).
```
Comment (stickies-v) [inline doc/release-notes.md:92]:
```suggestion
  relaxed or disabled (e.g. with `-acceptnonstdtxn=1`). (#28948)
```

File: doc/release-notes.md (hunk 6 above)
Comment (stickies-v) [reply doc/release-notes.md:140]: Incorrect pull number

```suggestion
- Improved handling of empty settings.json files (#29144)
```

**Review by stickies-v — COMMENTED** [review]
LGTM 4681f554bc6eb6f603512575de19bc52d08813e9 with a few release note nits

- backports mostly clean, 753c68dc0f02644f223d407b33000dce7a763056 being the exception because of LLVM version bump
- release notes look good and correspond with devwiki, quickly went through merges since v26.0 and couldn't see any major omissions
- i'm getting the same manpages output

**Comment by fanquake** [comment]
@vostrnad @stickies-v thanks, combined suggestions.

**Review by stickies-v — APPROVED** [review]
ACK 910e3e8728b258a38d38f8f9ddf6b23406e8d5ce - no changes except addressing nits

**Comment by achow101** [comment]
ACK 910e3e8728b258a38d38f8f9ddf6b23406e8d5ce

**Pull request merged**